### PR TITLE
155 deal with anaconda licensing changes

### LINKED ISSUE #155: Deal with Anaconda licensing changes
Need to move away from using the default conda channel to comply with new licensing. Need to update the example environment, docs, and workflows.

### PR DESCRIPTION
**Pull request by the-other-james**
Closes #155 

Switches to miniforge and removes the default channel for the workflows and example Docker environment. Tested by checking workflow output and on my local machine for Docker.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -18,11 +18,17 @@ jobs:
...
+      run: |
+        conda config --remove channels defaults
+        pylint onair plugins test
     - name: Check Formatting
       shell: bash -l {0}
-      run: black --check .
\ No newline at end of file
+      run: black --check .
+    - name: Confirm no defaults
+      run: conda config --show channels
```
Comment (asgibson) [inline .github/workflows/lint.yml:34]: Does this just offer an output and not really do any actual checks?

File: .pylintrc
Comment (asgibson) [inline .pylintrc:1]: Not sure why this is stating updated. Not sure this edit needs to be included.

File: Dockerfile [hunk 2]
```diff
@@ -57,13 +57,13 @@ USER onair_dev
...
-  bash ~/miniconda.sh -b -u -p $CONDA_DIR && \
-  rm -rf ~/miniconda.sh
+  wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh -O ~/miniforge.sh && \
+  bash ~/miniforge.sh -b -u -p $CONDA_DIR && \
+  rm -rf ~/miniforge.sh
 ENV PATH=$CONDA_DIR/bin:$PATH
 
 # Make OnAir requirements file accessible by onair_dev user
-COPY environment.yml /home/onair_dev/environment.yml
+COPY environment_dev.yml /home/onair_dev/environment.yml
```
Comment (asgibson) [inline Dockerfile:66]: environment_dev.yml includes Black and Pylint. Did we want those required for the container?

File: .pylintrc [hunk 3]
```diff
@@ -1,3 +1,3 @@
 [MAIN]
 
-exit-zero=True
\ No newline at end of file
+exit-zero=True
```
Comment (asgibson) [inline .pylintrc:3]: Not sure why this is stating updated. Not sure this edit needs to be included.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (the-other-james) [reply .github/workflows/lint.yml:34]: Yes, I looked at the output manually.

File: Dockerfile (hunk 2 above)
Comment (the-other-james) [reply Dockerfile:66]: I'm using it for my dev environment. https://github.com/nasa/OnAIR/issues/157

**Pull request merged**